### ISSUE TITLE
Error when running credit mining: outgoing port not available

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#2741]
Got this error reported today:

```
Traceback (most recent call last):
File "/media/Curent/WORK/Tribler/tribler/TriblerGUI/core_manager.py", line 128, in on_received_state
raise RuntimeError(state['last_exception'])
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 417, in fireEvent
DeferredList(beforeResults).addCallback(self._continueFiring)
File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 317, in addCallback
callbackKeywords=kw)
File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 306, in addCallbacks
self._runCallbacks()
File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 588, in _runCallbacks
current.result = callback(current.result, *args, **kw)
--- <exception caught here> ---
File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 430, in _continueFiring
callable(*args, **kwargs)
File "/media/Curent/WORK/Tribler/tribler/TriblerGUI/core_manager.py", line 58, in start_tribler
session.start()
File "/media/Curent/WORK/Tribler/tribler/Tribler/dispersy/util.py", line 46, in helper
return blockingCallFromThread(reactor, func, *args, **kargs)
File "/media/Curent/WORK/Tribler/tribler/Tribler/dispersy/util.py", line 225, in blockingCallFromThread
return f(*args, **kwargs)
File "/media/Curent/WORK/Tribler/tribler/Tribler/Core/Session.py", line 540, in start
startup_deferred = self.lm.register(self, self.sesslock)
File "/media/Curent/WORK/Tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py", line 185, in register
self.init()
File "/media/Curent/WORK/Tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py", line 331, in init
self.boosting_manager = BoostingManager(self.session)
File "/media/Curent/WORK/Tribler/tribler/Tribler/Core/CreditMining/BoostingManager.py", line 99, in __init__
{'share_mode_target': self.settings.share_mode_target})
exceptions.KeyError: 'outgoing_port'
```

@ardhipoetra can you investigate this?

**Comment by ardhipoetra** [comment]
Welp, this is weird. I never set `outgoing_port` in the credit mining settings. Even if I do, `outgoing_port` is a valid key in either new or old version of libtorrent. Does this error happen frequently?

**Comment by devos50** [comment]
@ardhipoetra nope, this is the first time I saw it.

**Comment by devos50** [comment]
Fixed by #3390
